src/v/cluster/topics_frontend.cc

```diff
@@ -195,10 +198,46 @@ ss::future<std::vector<topic_result>> topics_frontend::update_topic_properties(
       });
 }
 
+ss::future<std::error_code> topics_frontend::do_update_data_policy(
+  topic_properties_update& update, model::timeout_clock::time_point timeout) {
+    // Alter config rewrites all properties, so we need to delete dp if op is
+    // set and vaule is nullopt
+    bool is_convert_from_set_to_remove = false;
+    if (
+      update.properties.data_policy.op == incremental_update_operation::set
+      && !update.properties.data_policy.value.has_value()) {
+        update.properties.data_policy.op = incremental_update_operation::remove;
+        is_convert_from_set_to_remove = true;
+    }
+
+    switch (update.properties.data_policy.op) {
+    case incremental_update_operation::set:
+        co_return co_await _dp_frontend.local().create_data_policy(
+          update.tp_ns, update.properties.data_policy.value.value(), timeout);
+    case incremental_update_operation::remove: {
+        auto res = co_await _dp_frontend.local().clear_data_policy(
```

Comment:
meybe it would be easier to make `clear_data_policy` idempotent. i.e. not returning an error when deleted policy is not present ?

Reply:
@dotnwat What do you think about it?
I do not have strong opinion

Reply:
https://github.com/vectorizedio/redpanda/pull/2167#discussion_r698976991

Reply:
I don't have a strong opinion, but it seems like returning an error is not in conflict with being idempotent: if you delete something that doesn't exist and get back an error=ENOENT then the target semantic of "not returning an error when policy is not present" can be achieved by ignoring ENOENT errors.

Reply:
It can be problem if we delete this property with another in one request
Request will be failed in remove data policy and other changes will not be applied

Reply:
I guess it may be a problem it will render `ALTER_CONFIG` api not usable. When one would like to set f.e. retention properties and there is no data policy alter config handler will try to reset it. This will result in an error since there is no data policy already

Reply:
@VadimPlh please test alter config and incremental alter config and make sure they both work as expected. For example, I think that @mmaslankaprv is saying that ALTER_CONFIG will reset data policy unless its provided each time the configuration is provided. That is one reason why we might want to _not_ support non-incremental alter config.

Reply:
Current behavior:

* alter config with data policy specific:
* alter config without data policy specific:
* incremental alter config with data policy specific:
* incremental alter config without data policy specific:

Reply:
- alter config with data policy specific:
If data-policy exist - will return error, if updates contains another properties will return error and does not change anything
If data-policy does not exist - set it. Set another properties which does not exist in request to default

- alter config without data policy specific:
rewrite properties. Does not return an error, if property existed before.
It set default for all properties wich are not in update request( So it will disable data-policy)

- incremental alter config with data policy specific:
set: Set data-olicy if it does not exist, else return error
remove: Remove data-policy if it exists, else return error.
Does not change properties which does not exits in request

- incremental alter config without data policy specific:
set: rewrite property. Does not return error
remove: remove property (set it to default). Does not return error 
Does not change properties which does not exits in request